src/binder_voicecall.c

```diff
@@ -300,6 +300,54 @@ binder_voicecall_info_ext_new(
     return call;
 }
 
+static
+const BinderVoiceCallInfo*
+binder_voicecall_list_find_call_with_id(
+    GSList* list,
+    guint call_id)
+{
+    for (GSList* l = list; l; l = l->next) {
+        const BinderVoiceCallInfo* call = l->data;
+
+        if (call->oc.id == call_id) {
+            return call;
+        }
+    }
+    return NULL;
+}
+
+static
+GSList*
+binder_voicecall_merge_call_lists(
+    BinderVoiceCall* self,
+    GSList* new_list,
+    gboolean add_ext)
+{
+    /*
+     * Since IRadio and ext may have different lists of calls,
+     * attempt to merge the ongoing calls initiated by the other
+     * component before current calls list is going to
+     * be replaced by the new list.
+     */
+    if (!self->ext)
+        return new_list;
+
+    for (GSList* l = self->calls; l; l = l->next) {
+        const BinderVoiceCallInfo* call = l->data;
+
+        if (!!call->ext == add_ext) {
+            if (binder_voicecall_list_find_call_with_id(new_list, call->oc.id))
+                continue;
+
+            new_list = g_slist_insert_sorted(new_list,
+                g_slice_dup(BinderVoiceCallInfo, call),
+                binder_voicecall_info_compare);
```

Comment:
Is there a reason not to break the loop when the call is found? Just for the sake of sanity, I understand that in practice there will rarely be more than once call in the list.

Reply:
The way I thought it was to skip the particular call that exists in both lists, but still check if there are others that are not. Do you suggest to exit early if a single call is found in both lists?

Reply:
Ah, you're right.